File: post/src/main/java/ripoff/facebook/post/dto/PostCreationRequest.java
package ripoff.facebook.post.dto;

import lombok.*;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;
import java.util.Set;

@Data
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PostCreationRequest {

    private Long userId;
    private String content;
    private VisibilityGroupType visibilityGroupType;
    private Long visibilityGroupId;
    private Set<Long> visibilityUsersId;
}

File: frontend/src/app/application/feature/post/create-post-page/post-creation-request.ts
export class PostCreationRequest {
  private content: string;
  private visibilityGroupType: string;
  private visibilityUsersId: number[];

  constructor(content: string, groupType: string, visibilityUsersId?: number[]) {
    this.content = content;
    this.visibilityGroupType = groupType;
    if(visibilityUsersId !== undefined){
      this.visibilityUsersId = visibilityUsersId;
    } else {
      this.visibilityUsersId = [];
    }
  }

}
